widgetssdk/src/test/java/com/glia/widgets/core/survey/domain/GliaSurveyAnswerUseCaseTest.java

```diff
@@ -0,0 +1,676 @@
+package com.glia.widgets.core.survey.domain;
+
+import static junit.framework.TestCase.assertNull;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.glia.androidsdk.engagement.Survey;
+import com.glia.widgets.core.survey.GliaSurveyRepository;
+import com.glia.widgets.survey.QuestionItem;
+import com.glia.widgets.survey.SurveyValidationException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+public class GliaSurveyAnswerUseCaseTest {
+
+    private GliaSurveyAnswerUseCase gliaSurveyAnswerUseCase;
+
+    @Before
+    public void setUp() {
+        GliaSurveyRepository repository = mock(GliaSurveyRepository.class);
+        gliaSurveyAnswerUseCase = new GliaSurveyAnswerUseCase(repository);
+    }
+
+    // Tests for a single question validation
+
+    @Test
+    public void validate_succeeds_whenQuestionOptionalTypeTextAnswerNull() {
+        Survey.Question question = mock(Survey.Question.class);
+        when(question.isRequired()).thenReturn(false);
+        when(question.getType()).thenReturn(Survey.Question.QuestionType.TEXT);
+
+        Survey.Answer answer = null;
+        QuestionItem item = new QuestionItem(question, answer);
+        try {
+            gliaSurveyAnswerUseCase.validate(item);
+        } catch (Exception e) {
+            Assert.fail("Should not have thrown any exception");
+        }
```

Comment:
NIT: why do you wrap the tests in exception. Is it something from styling/guides? I mean the test will fail even if the method just throws the exception

Reply:
@gugalo Yes, the test will fail even if the method just throws an exception. The difference is, that with `Assert.fail` the test will fail with a readable assertion exception, and without it, the test will fail because of the crash.